### PR TITLE
[11.0][ADD] maintenance_equipment_tags

### PR DESCRIPTION
**Pull request by jarroyomorales**
Equipment already has a category field, but these tags allow you to have more flexibility to make groups.

### CODE REVIEW

File: maintenance_equipment_tags/views/maintenance_equipment_tag.xml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2019 Creu Blanca
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+
+<odoo>
+
+    <record model="ir.ui.view" id="maintenance_equipment_tag_form_view">
```
Comment (astirpe) [inline maintenance_equipment_tags/views/maintenance_equipment_tag.xml]: Put `id=...` before `model=...`, as per the OCA guidelines

File: maintenance_equipment_tags/views/maintenance_equipment_tag.xml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                <sheet>
+                    <group>
+                        <field name="name"/>
+                    </group>
+                </sheet>
+            </form>
+        </field>
+    </record>
+
+    <record model="ir.ui.view" id="maintenance_equipment_tag_tree_view">
```
Comment (astirpe) [inline maintenance_equipment_tags/views/maintenance_equipment_tag.xml]: Put `id=...` before `model=...`, as per the OCA guidelines

File: maintenance_equipment_tags/views/maintenance_equipment_tag.xml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        <field name="name">maintenance.equipment.tag.tree (in maintenance_equipment_tags)</field>
+        <field name="model">maintenance.equipment.tag</field>
+        <field name="arch" type="xml">
+            <tree>
+                <field name="name"/>
+            </tree>
+        </field>
+    </record>
+
+    <record model="ir.actions.act_window" id="maintenance_equipment_tag_act_window">
```
Comment (astirpe) [inline maintenance_equipment_tags/views/maintenance_equipment_tag.xml]: Put `id=...` before `model=...`, as per the OCA guidelines

File: maintenance_equipment_tags/views/maintenance_equipment_tag.xml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        <field name="name">Maintenance Equipment Tags</field>
+        <field name="res_model">maintenance.equipment.tag</field>
+        <field name="view_mode">tree,form</field>
+        <field name="domain">[]</field>
+        <field name="context">{}</field>
+    </record>
+
+    <record model="ir.ui.menu" id="maintenance_equipment_tag_menu">
```
Comment (astirpe) [inline maintenance_equipment_tags/views/maintenance_equipment_tag.xml]: Put `id=...` before `model=...`, as per the OCA guidelines

File: maintenance_equipment_tags/security/ir.model.access.csv [hunk 5]
```diff
@@ -0,0 +1,2 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
```
Comment (astirpe) [inline maintenance_equipment_tags/security/ir.model.access.csv:1]: You are missing something here. The group base.group_user has only red access, which is fine. But nobody has rights to modify, create and delete tags. I think any maintenance manager should have such rights.

File: maintenance_equipment_tags/views/maintenance_equipment.xml [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+            <xpath expr="//div[hasclass('oe_title')]" position="after">
+                <field name="tag_ids" widget="many2many_tags"
+                       options="{'color_field': 'color'}"
+                       placeholder="e.g. Computers"
+                />
+            </xpath>
+        </field>
+    </record>
+
+    <record model="ir.ui.view" id="hr_equipment_view_kanban">
```
Comment (astirpe) [inline maintenance_equipment_tags/views/maintenance_equipment.xml]: Put `id=...` before `model=...`, as per the OCA guidelines

File: maintenance_equipment_tags/views/maintenance_equipment.xml [hunk 7]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2019 Creu Blanca
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+
+<odoo>
+
+    <record model="ir.ui.view" id="hr_equipment_view_form">
```
Comment (astirpe) [inline maintenance_equipment_tags/views/maintenance_equipment.xml]: Put `id=...` before `model=...`, as per the OCA guidelines

File: maintenance_equipment_tags/views/maintenance_equipment_tag.xml [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+    <record model="ir.ui.view" id="maintenance_equipment_tag_form_view">
+        <field name="name">maintenance.equipment.tag.form (in maintenance_equipment_tags)</field>
+        <field name="model">maintenance.equipment.tag</field>
+        <field name="arch" type="xml">
+            <form>
+                <sheet>
+                    <group>
+                        <field name="name"/>
+                    </group>
```
Comment (astirpe) [inline maintenance_equipment_tags/views/maintenance_equipment_tag.xml:16]: Can you also allow the user to select a different color?

File: maintenance_equipment_tags/models/maintenance_equipment_tag.py [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+
+    def get_default_color_value(self):
+        return randint(1, 15)
+
+    name = fields.Char(string="Equipment Tag", required=True)
+    color = fields.Integer(
+        string='Color Index',
+        default=lambda self: self.get_default_color_value()
+    )
+    equipment_ids = fields.Many2many(
```
Comment (astirpe) [inline maintenance_equipment_tags/models/maintenance_equipment_tag.py:21]: I don't think this field is necessary here...
In any case, `string='Employees'` is not appropriate.

**Review by astirpe — CHANGES_REQUESTED** [review]
Only code review. Not tested.

File: maintenance_equipment_tags/security/ir.model.access.csv (hunk 5 above)
Comment (jarroyomorales) [reply maintenance_equipment_tags/security/ir.model.access.csv:1]: Added rule: `access_maintenance_equipment_man,access_maintenance_equipment_tag,model_maintenance_equipment_tag,maintenance.group_equipment_manager,1,1,1,1`

File: maintenance_equipment_tags/models/maintenance_equipment_tag.py (hunk 9 above)
Comment (jarroyomorales) [reply maintenance_equipment_tags/models/maintenance_equipment_tag.py:21]: Sorry, copy-pasted from hr and didnt change it. I think it doesnt do any bad and can be useful if someone wants to access equipment from tags

File: maintenance_equipment_tags/security/ir.model.access.csv [hunk 10]
```diff
@@ -0,0 +1,3 @@
...
+access_maintenance_equipment_tag,access_maintenance_equipment_tag,model_maintenance_equipment_tag,base.group_user,1,0,0,0
+access_maintenance_equipment_man,access_maintenance_equipment_tag,model_maintenance_equipment_tag,maintenance.group_equipment_manager,1,1,1,1
```
Comment (astirpe) [inline maintenance_equipment_tags/security/ir.model.access.csv]: Here the name of the record is the same as the previous one: `access_maintenance_equipment_tag`. Should be different.

**Review by astirpe — CHANGES_REQUESTED** [review]

**Review by astirpe — APPROVED** [review]

**Review by etobella — APPROVED** [review]

**Review by JordiBForgeFlow — APPROVED** [review]

**Pull request merged**